### DISCUSSION TITLE
0.38.0-a1

### DISCUSSION THREAD

**Discussion by rmorshea** [idom-team/idom#719] [Announcements]
## What's Changed
* Rework How IDOM Integrates With Servers by @rmorshea in https://github.com/idom-team/idom/pull/703

**Full Changelog**: https://github.com/idom-team/idom/compare/0.37.2...0.38.0-a1

<hr /><em>This discussion was created from the release <a href='https://github.com/idom-team/idom/releases/tag/0.38.0-a1'>0.38.0-a1</a>.</em>

**Comment by rmorshea** [comment]
@acivitillo, this pre-release includes a lot of changes. You should read the [changelog](https://idom-docs.herokuapp.com/docs/about/changelog.html) for a summary, as well as the updated section on [running IDOM](https://idom-docs.herokuapp.com/docs/guides/getting-started/running-idom.html) using the new interfaces.

Notably though, this release includes hooks for each server implementation that expose underlying websockets and requests. At a minimum, all implementations expose a `use_scope` hook, but many provide `use_request` or `use_websocket`. For example [`idom.server.starlette.use_websocket`](https://idom-docs.herokuapp.com/docs/_auto/apis.html#idom.server.starlette.use_websocket).

Any feedback on the documentation or the changes themselves would be helpful before releasing 0.38.0.

**Comment by sylvoslee** [comment]
```
from fastapi import FastAPI

from idom import component, html
from idom.server.fastapi import configure, Options


@component
def HelloWorld():
    return html.h1("Hello, world!")


app = FastAPI()
configure(app, HelloWorld, Options(url_prefix="/test"))

uvicorn main:app
```
On my windows computer, I get the following error：
```

Traceback (most recent call last):
  File "c:\users\envs\debug\lib\asyncio\selector_events.py", line 862, in _read_ready__data_received
    self._protocol.data_received(data)
  File "c:\users\envs\debug\lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 131, in data_received
    self.handle_upgrade()
  File "c:\users\envs\debug\lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 160, in handle_upgrade
    protocol = self.ws_protocol_class(
  File "c:\users\envs\debug\lib\site-packages\uvicorn\protocols\websockets\websockets_impl.py", line 62, in __init__
    super().__init__(
  File "c:\users\envs\debug\lib\site-packages\websockets\server.py", line 103, in __init__
    super().__init__(**kwargs)
TypeError: __init__() got an unexpected keyword argument 'logger'
```

**Reply by rmorshea** [reply]
I'm not observing that locally on my end. I'm guessing it has to do with some mismatched versions in your dependencies given where the error isn't arising from within IDOM and that everything is passing in CI.

**Reply by rmorshea** [reply]
Haven't looked into how to resolve yet, but [this issue](https://github.com/encode/uvicorn/issues/1320) looks similar.

**Reply by sylvoslee** [reply]
After upgrading the websockets library, the problem is solved, thanks
